sql/core/src/main/scala/org/apache/spark/sql/util/QueryExecutionListener.scala

```diff
@@ -109,6 +109,16 @@ class ExecutionListenerManager private[sql](session: SparkSession, loadExtension
     listenerBus.removeAllListeners()
   }
 
+  /**
+   * Removes [[ExecutionListenerBus]] from LiveListenerBus.queues. people should call this function
+   * when the manager belongs session was removed.
+   */
+  @DeveloperApi
+  def clearListenerBus(): Unit = {
```

Comment:
Does the user need to manually clear these listener event?

Reply:
I think it is a good idea to remove the listeners automatically. If I understand correctly, Spark now has no method to destroy a SparkSession, user manager session lifecycle by themself, so we can't remove the listener automatically. Maybe we can put clear listener behavior in `SparkSession.clearActiveSession()`, but i think it's not a good idea.